src/pwa/components/Link/index.js

```diff
@@ -128,16 +133,15 @@ class Link extends Component {
       eventCategory,
       eventAction,
     } = this.props;
-    setTimeout(
-      () =>
-        routeChangeRequested({
-          selectedItem: { type, id, page },
-          context,
-          method,
-          event: { category: eventCategory, action: eventAction },
-        }),
-      100,
-    );
+    setTimeout(() => {
+      sendEvent({ category: eventCategory, action: eventAction });
+      
+      routeChangeRequested({
+        selectedItem: { type, id, page },
+        context,
+        method,
+      });
+    }, 100);
```

Comment:
We are sending the event before the routeChange is succeed. The GA library will take the current url and page title, instead of the ones the user is moving to.

What would be the best solution here? current page or next page?

*Even if we turn it around, I think we may still have a problem because it takes a while for the browser to change the url and page-title.*

Reply:
This is more of a question than a requested change.

Reply:
After talk with @orballo, I think sending the events **before** request a route change is better solution, as we are tracking user actions in the current page (the route change would be the consequence of that action).

Reply:
Ok, great then.